src/computability/turing_machine.lean

```diff
@@ -634,11 +1090,21 @@ local notation `stmt₀` := TM0.stmt Γ
 parameters (M : Λ → stmt₁)
 include M
 
+/-- The base machine state space is a pair of an `option stmt₁` representing the current program
+to be executed, or `none` for the halt state, and a `σ` which is the local state (stored in the TM,
+not the tape). Because there are an infinite number of programs, this state space is infinite, but
+for a finitely supported TM1 machine and a finite type `σ`, only finitely many of these states are
+reachable. -/
+@[nolint unused_arguments]
```

Comment:
Please explain which arguments are unused, and why they need to be unused arguments.